### PR TITLE
Use the proper netlink buffer size with large kernel pages

### PR DESCRIPTION
**Pull request by refi64**
The recommended netlink buffer size is based on the system's page size, which means that the current size is far too small for systems with 16k or 64k pages, such as Asahi Linux or RHEL's kernel-64k for ARM64. On these systems, the server fails to start with errors like this:

Error: Decode error occurred: invalid netlink buffer: length field says 1444 the buffer is 1260 bytes long

Instead, follow the kernel's own netlink docs to compute the buffer size. The approach here matches the approach merged into Chromium recently:

https://chromium-review.googlesource.com/c/chromium/src/+/4312885

### CODE REVIEW

File: netlink-request/src/lib.rs
Comment (bschwind) [inline netlink-request/src/lib.rs:31]: Do you think we should use a default here instead of double unwrapping?

File: netlink-request/src/lib.rs
Comment (bschwind) [inline netlink-request/src/lib.rs:26]: Not a strong preference but this could just be called `max_netlink_buffer_length()` and the function below could be `max_genl_payload_length()`

File: netlink-request/src/lib.rs
Comment (bschwind) [inline netlink-request/src/lib.rs:29]: Small nitpick

```suggestion
        // size of the CPU architecture, whichever is bigger."
```

**Review by bschwind — APPROVED** [review]
Good catch here! The code looks good, I just have some mostly trivial comments on naming and such.

File: netlink-request/src/lib.rs
Comment (refi64) [reply netlink-request/src/lib.rs:35]: This shouldn't ever fail unless something has gone horribly wrong, so it's probably safer to just unwrap; I added a comment clarifying this.

File: netlink-request/src/lib.rs
Comment (refi64) [reply netlink-request/src/lib.rs]: Updated, also did some variable renames to match.

**Review by bschwind — APPROVED** [review]
LGTM++, I'll wait for at least another review before merging.

File: netlink-request/Cargo.toml
Comment (strohel) [inline netlink-request/Cargo.toml:11]: I see that nix has many features, all of them enabled by default. Looks like we can only enable nix cargo feature called `feature` (:srug:) to save a bit of compilation time?

**Review by strohel — CHANGES_REQUESTED** [review]
Looking great! Just one little suggestion wrt dependencies.

I guess the `sysconf()` call is rather cheap and we don't have to think about caching it (beyond local variable caching already present in this PR)?

**Review by mcginty — APPROVED** [review]
Nice comments, nice fix!

+1 for @strohel's suggestion of not using default features in `nix`.

On my machine, `sysconf` is <1ms but is called >200 times given that it's called in a pretty low-level function. I don't think it's a bad idea just from a cleanliness perspective to throw it in a [`OnceCell`](https://github.com/matklad/once_cell), especially since Rust will likely stabilize a similar API in `std` soon enough.

**Review by mcginty — APPROVED** [review]
`OnceCell` change looks good!

**Pull request merged**
